src/github/tests/test_utils.rs

```diff
@@ -0,0 +1,193 @@
+    teams: Vec<TeamData>,
+}
+
+impl DataModel {
+    pub fn user(&mut self, name: &str) -> usize {
+        let github_id = self.people.len();
+        self.people.push(Person {
+            name: name.to_string(),
+            email: Some(format!("{name}@rust.com")),
+            github_id,
+        });
+        github_id
+    }
+
+    pub fn gh_model(&self) -> GitHubModel {
+        GitHubModel {
+            users: self
+                .people
+                .iter()
+                .map(|user| (user.github_id, user.name.clone()))
+                .collect(),
+            owners: Default::default(),
+            teams: self
+                .teams
+                .clone()
+                .into_iter()
+                .enumerate()
+                .map(|(id, team)| api::Team {
+                    id: Some(id),
+                    name: team.name.clone(),
+                    description: None,
+                    privacy: TeamPrivacy::Closed,
+                    slug: team.name,
+                })
+                .collect(),
+        }
+    }
+
+    pub fn team(&mut self, team: TeamDataBuilder) {
+        let team = team.build().expect("Cannot build team");
+        self.teams.push(team);
+    }
+
+    pub fn diff_teams(&self, github: GitHubModel) -> Vec<TeamDiff> {
+        let teams = self.teams.iter().map(|r| r.to_data()).collect();
+        let repos = vec![];
+
+        let read = Rc::new(github);
+        let sync = SyncGitHub::new(read, teams, repos).expect("Cannot create SyncGitHub");
+        sync.diff_teams().expect("Cannot diff teams")
+    }
+}
+
+#[derive(Clone, Builder)]
+#[builder(pattern = "owned")]
+pub struct TeamData {
+    #[builder(default = "TeamKind::Team")]
+    kind: TeamKind,
+    name: String,
+    #[builder(default)]
+    gh_teams: Vec<GitHubTeam>,
+}
+
+impl TeamData {
+    pub fn new(name: &str) -> TeamDataBuilder {
+        TeamDataBuilder::default().name(name.to_string())
+    }
+
+    fn to_data(&self) -> rust_team_data::v1::Team {
+        let TeamData {
+            name,
+            kind,
+            gh_teams,
+        } = self.clone();
+        rust_team_data::v1::Team {
+            name: name.clone(),
+            kind,
+            subteam_of: None,
+            members: vec![],
+            alumni: vec![],
+            github: (!gh_teams.is_empty()).then(|| TeamGitHub { teams: gh_teams }),
+            website_data: None,
+            discord: vec![],
+        }
+    }
+}
+
+impl TeamDataBuilder {
+    pub fn gh_team(mut self, name: &str, members: &[usize]) -> Self {
+        let mut gh_teams = self.gh_teams.unwrap_or_default();
+        gh_teams.push(GitHubTeam {
+            org: DEFAULT_ORG.to_string(),
+            name: name.to_string(),
+            members: members.to_vec(),
+        });
+        self.gh_teams = Some(gh_teams);
+        self
+    }
+}
+
+/// Represents the state of GitHub repositories, teams and users.
+#[derive(Default, Clone)]
+pub struct GitHubModel {
```

Comment:
Nit: `GithubMock` or similar would be clearer IMO - I had to read this struct definition to know what this was for, but a different name would have made that clearer I think.